docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -139,11 +143,80 @@ the mempool, regardless of their classes, will have a *partial order*.
 This means that some pairs of
 transactions are comparable and, thus, have and order, while others not.
 
+#### Network-wide consistency
+
+The properties presented so far may be interpreted as per-node properties.
+However, we need to define some network-wide properties in order for a mempool QoS implementation
+to be useful and predictable for the whole appchain network.
+These properties are expressed in terms of consistency of the information, configuration and behaviour
+across nodes in the network.
+
+:parking: _Property_ **Consistent transaction classes**: For any transaction `tx`,
+and any two correct nodes $p$ and $q$ that receive `tx` for the first time,
+$p$ and $q$ MUST have the same the set of transaction classes and their relative priority and configuration,
+as long as `tx` has not been included in a block.
+
+> TODO: Do we really need to require that the `tx` hasn't been decided? Isn't it adding complexity for nothing?
```

Comment:
This property needs to cover the cases when the node actually makes a CheckTx request to the app. This happens when the tx arrives for the first time (meaning tx first passes the cache check) and when rechecking. So both scenarios are before the tx is decided in a block. We have to keep in mind that a tx may be decided in a block and still be in the cache. And that a node may never receive a tx from a peer, thus not having the tx in the mempool, but still add the tx to the cache (when updating the mempool with a block containing the tx). So I'd say that the property is correct as it is.

Reply:
Let's maybe briefly discuss it verbally on Mon